lesson 9 classes

mariusamsarius opened the pull request:
add my homework for lesson 9

Commit:
Adding lesson 7 assignment

Commit:
lesson 7 decorators

Commit:
lesson 7 decorators

Commit:
lesson 9 classes

lorena91 commented on lesson_9/lesson9_classes.py:
```diff
@@ -0,0 +1,62 @@
+
+    def category_check(self):
+        if self.category is 'Laptop':
+            return f'Category: Computers'
```
You could also handle the else case

lorena91 commented on lesson_9/lesson9_classes.py:
```diff
@@ -0,0 +1,62 @@
+hp = Laptop('Laptop', 'HP', 'Pavilion', 2, '4999 RON', '15,6"', '4K')
+asus = Laptop('Laptop', 'Asus', '8565U', 4, '2999 RON','14"', 'Ultraportable')
+dell = Laptop('Laptop', 'Dell', 'Alienware', 0, '9999 RON','17"', 'Gaming')
+
```
I think there should be 5 instances

mariusamsarius replied:
added one more instance

lorena91 reviewed (commented):
Looking good and clean! :+1:

Commit:
update

Commit:
Merge branch 'lesson_7_dragla_marius' of https://github.com/mariuspodean/CJ-PYTHON-01 into lesson_9_dragla_marius

Commit:
pull l7

Commit:
lesson 9 update

Liitle commented:
Hi, Please remove the files from lesson_7 :smiley:. Also, you can create a file in the root of the project named `.gitignore` and add in that file `.idea` this operation will exclude PyCharm config files `.idea`.
If you want you can also exclude `.idea` files manually. 

If you are having trouble making the changes let me know and I will help

Liitle commented on lesson_9/dragla_marius/lesson9_classes.py:
```diff
@@ -0,0 +1,64 @@
+            return 'Availability: Out of stock'
+
+    def price(self):
+        return self.price
```
Usually, the methods need to contain some form of functionality. If you just return the value the function call is similar to just reading the attribute from the instance.

Laptop.price will be the same as self.price. Also having the same name for methods and attributes is not good practice. Try to add some functionality to the method and change the name :smile:

mariusamsarius replied:
added functionality to convert price to euro

Liitle commented on lesson_9/dragla_marius/lesson9_classes.py:
```diff
@@ -0,0 +1,64 @@
+    def description(self):
+        return f'{self.category} {self.brand} {self.model} {self.display}'
+
+    def special_feature(self):
```
same as the comment above

mariusamsarius replied:
done

Liitle reviewed (commented):
Nice work, clean code and to the point. Some small changes are required, see the comments bellow

Commit:
lesson 9 update

Commit:
lesson 9 changes

mariusamsarius commented:
> Hi, Please remove the files from lesson_7 smiley. Also, you can create a file in the root of the project named `.gitignore` and add in that file `.idea` this operation will exclude PyCharm config files `.idea`.
> If you want you can also exclude `.idea` files manually.
> 
> If you are having trouble making the changes let me know and I will help

@Liitle - I have deleted lesson 7 and added the .idea line in gitignore, but I am not sure how to delete the .idea folder on my branch. If I do a pull from git to my local project, I cannot see the .idea folder anywhere. Can you please advise how I can remove it from the git branch?
